ssc/cmod_singleowner.cpp

```diff
@@ -1064,31 +1064,39 @@ class cm_singleowner : public compute_module
-			nameplate1 = as_number("om_batt_nameplate");
-            if (as_integer("en_batt") == 1 || as_integer("en_standalone_batt") == 1 || as_integer("en_wave_batt") == 1)
+        std::vector<double> battery_discharged(nyears,0);
+        std::vector<double> fuelcell_discharged(nyears,0);
+
+        if (add_om_num_types > 0) //PV Battery
+        {
+            escal_or_annual(CF_om_fixed1_expense, nyears, "om_batt_fixed_cost", inflation_rate, 1.0, false, as_double("om_fixed_escal") * 0.01);
+            escal_or_annual(CF_om_production1_expense, nyears, "om_batt_variable_cost", inflation_rate, 0.001, false, as_double("om_production_escal") * 0.01); //$/MWh
+            escal_or_annual(CF_om_capacity1_expense, nyears, "om_batt_capacity_cost", inflation_rate, 1.0, false, as_double("om_capacity_escal") * 0.01);
+            nameplate1 = as_number("om_batt_nameplate");
+            if (as_integer("en_batt") == 1 || as_integer("en_standalone_batt") == 1)
                 battery_discharged = as_vector_double("batt_annual_discharge_energy");
-		}
-		if (add_om_num_types > 1) // PV Battery Fuel Cell
-		{
-			escal_or_annual(CF_om_fixed2_expense, nyears, "om_fuelcell_fixed_cost", inflation_rate, 1.0, false, as_double("om_fixed_escal")*0.01);
-			escal_or_annual(CF_om_production2_expense, nyears, "om_fuelcell_variable_cost", inflation_rate, 0.001, false, as_double("om_production_escal")*0.01);
-			escal_or_annual(CF_om_capacity2_expense, nyears, "om_fuelcell_capacity_cost", inflation_rate, 1.0, false, as_double("om_capacity_escal")*0.01);
-			nameplate2 = as_number("om_fuelcell_nameplate");
-            fuelcell_discharged = as_vector_double("fuelcell_annual_energy_discharged");
-		}
+        }
+        if (battery_discharged.size() == 1) { // ssc #992
+            double first_val = battery_discharged[0];
+            battery_discharged.resize(nyears, first_val);
+        }
+        if (battery_discharged.size() != nyears)
+            throw exec_error("singleowner", util::format("battery_discharged size (%d) incorrect",(int)battery_discharged.size()));
 
+        if (add_om_num_types > 1)
+        {
+            escal_or_annual(CF_om_fixed2_expense, nyears, "om_fuelcell_fixed_cost", inflation_rate, 1.0, false, as_double("om_fixed_escal")*0.01);
+            escal_or_annual(CF_om_production2_expense, nyears, "om_fuelcell_variable_cost", inflation_rate, 0.001, false, as_double("om_production_escal")*0.01);
+            escal_or_annual(CF_om_capacity2_expense, nyears, "om_fuelcell_capacity_cost", inflation_rate, 1.0, false, as_double("om_capacity_escal")*0.01);
+            nameplate2 = as_number("om_fuelcell_nameplate");
+            fuelcell_discharged = as_vector_double("fuelcell_annual_energy_discharged");
+        }
+        if (fuelcell_discharged.size()== 1) { // ssc #992
+            double first_val = fuelcell_discharged[0];
+            fuelcell_discharged.resize(nyears, first_val);
+         }
+        if (fuelcell_discharged.size() != nyears)
```

Comment:
The ssc test failures appear to be caused by this line. Fuel cell and battery use different length conventions for output variables: battery is nyears, fuel cell is nyears + 1. I believe battery to be correct, but can't find where we wrote down the convention of the financial models pre-pending year 0.

Reply:
@brtietz , the convention as far as I know is the the technology models start at year 1 and the financial models start at year zero. So, I have no idea about the cmod_fuelcell allocations:
![image](https://user-images.githubusercontent.com/6498311/218674607-f29bcc56-b223-4e5f-bac4-9f06a5810391.png)

I have updated all financial models to handle the fuel cell exception to the technology model rule in the last commit.